### ISSUE TITLE
Renaming files? A better way for archiving future

### ISSUE THREAD

**Issue by denemier** [Darakah/obsidian-gallery#8]
Hey, why aren't you naming fileinfo files better with the same name instead? think it as a alternative for xmp file format which is used for extended metadata.

**Currently:**
assets/galleryinfo/Img_11.md

**A Better Way:**
assets/food.jpg
->
assets/galleryinfo/food.md

**Comment by denemier** [comment]
its better to find files when we need them in future without any program for archiving safety i think

**Comment by Darakah** [comment]
The problem with this approach is that if more then 1 image use the same name (which can happen simply by the images being in different locations), then the info files will get overwritten. To add a number to the file name would require a scan of the info directory every time a new info file must be created which is rather better to avoid. In addition, as the file can be directly edited from the gallery, then there is no need to search for it. 

In short, this way is more user proof and faster. 

Would adding a button to open the info field directly work for you? that way you can jump to it from the side panel of the gallery and rename it to whatever you want there.

**Comment by denemier** [comment]
hey again, i was just checking something and found this:
1- if i add tags to a images info file
2- and rename this media file later in obsidian to file2.jpg
3-and see open the file again it doesnt show my tags anymore sync is kinda broken how can i avoid this is it a bug
and this is what i meant if it was broken i have lots of numeric files which i dont what file it belongs

**Comment by denemier** [comment]
and is it there duplicate files for 1 file after i rename it to something else, if so this would mean i can have more info files than the files related

**Comment by denemier** [comment]
by the way i am trying to combine all my photos archive with obsidian so i can mention any photo in any note, but then i needed a solution to tag my photos and search them later like
show photos #food captured in #sahara in 2021 (date)

normally people use xmp external metadata file along the original file because not every file format like has the ability to have metadata stored in them. xmp was invented because of this. which i dont like that much,
your format is much nicer compared xmp's xml file format.

but i need to be sure that this system works stable for tagging the files and protects files being deleted, or duplicated

**Comment by Darakah** [comment]
No, that shouldn't be happening.  There should be 1 info file per image (only 1 file).
Thanks for reporting this. 

## Renaming an Image:

if you rename an image, the link in the info file should update automatically (i.e. the thing between the `span`), BUT what you need to change automatically is the path written inside the `gallery-info` block of that info note since that will not change automatically (thus it will show as if the image doesn't exist when you open the info file again).

## File duplication:

I thought i found this duplicate info file problem.

Which version do you have?
Do you have sync on this vault?

**Comment by denemier** [comment]
v0.11.13 and i downloaded plugin today can you check my problem i recorded a video for it
https://user-images.githubusercontent.com/63566495/115233060-25e09f00-a0b3-11eb-9b69-d060e85c4cf5.mp4

**Comment by denemier** [comment]
when i change media file names in obsidian it definitely makes duplicate files or more files, i have disabled all other community plugins installed yours again had only 4 images and now i have 9 files some empty some duplicate

**Comment by denemier** [comment]
oh and i am not using any cloud or account like that its totaly on my local disk no sync whatsoever

**Comment by Darakah** [comment]
Hmm, thanks a lot for the example will look into it when I have some free time. 
Seems to be a problem with cache checking and update link. Will modify it to read each of the files instead ... should be a bit slower but more solid approach.

**Comment by denemier** [comment]
here another example screentop captured only this time pay attention that this time no rename or whatsoever
just looking image first than the info file and the info file content somehow gets deleted each time after few seconds as you can see in the video and when i delete everything let the gallery plugin start again makin files sometimes i see it starts from 11 not 1 is it there somehow cache or database you are using in the background i think


https://user-images.githubusercontent.com/63566495/115343082-0430f780-a147-11eb-9da3-edc58fdfde2f.mp4

**Comment by denemier** [comment]
and i have tested without keyboard plugged in, in case my keyboard is broken and presses a shortcut hiddenly.
but the problem is still valid without a keyboard attached to pc

**Comment by Darakah** [comment]
@denemier ,

Thanks a lot for your bug submission and detailed example.
I believe the following are fixed in `v0.5.5`:
- The info file getting overwritten
- Info file duplication on rename

Also changed the default naming behavior so that it uses the same name as the image otherwise it will append a count to it.

Thanks once again for your help,
in case any of these issues appear again feel free to re-open this issue.

**Comment by denemier** [comment]
wow this is exactly what i wanted, thanks for the same name behaviour,
just 1 thing to say, this is not must i can do it manuelly just asking for curiosity:

when we change the name of the file to something its automatically doing
- change the original jpg image file
- change all the linked mentions inside md note files, and same time because your plugin stores info files as md extension for medias obisidan changes between first span tags (gallery-span-info), because its a linked mention [[like-this]], this is very good for me,
but it's not changing the 
1-filename itself 
2-and not imgPath

my suggestions:
**1-**if there's a way to trigger a function to change filenames as well after obsidian changes a file for founded mention inside specified info files folder it will solve the problem, but i dont think the obsidian api is so advanced it may not be possible
**2-**look at the info file example below gallery-span-info is updated automatically by obsidian when image is renamed by titlebar, but imgPath= is not updated because its unlinked mention,

<span class='gallery-span-info'> [[wheelbarrow-new-name.jpg]] </span>
%% Place Tags Here %%
```gallery-info
imgPath=1-public/wheelbarrow.jpg
```

You can say it includes at beginnnig your gallery info path which you have specified in gallery plugin settings:
and that's true, but because we already programmatically know gallery info path as a parameter from plugin settings, there could be a better way i think: "to make gallery-info imgPath as a linked mention like the example below, or why do we even need gallery-info imgPath can't we just remove it "


<span class='gallery-span-info'> [[wheelbarrow-new-name.jpg]] </span>

%% Place Tags Here %%
```gallery-info
imgPath=[[wheelbarrow.jpg]]
```

the reason why i am so detailed about the renaming and file format thing is because this is not like a any other feauture it's backbone of the plugin, and i am aware changing the info file format every day is not acceptable for community because their files get so quickly old, but because you much thing, i would like to hear your opinios about if imgPath is unneccessary is or if it is a must, can we make it at least also a linkedmention inside brackets and get gallery info path from plugin setting when needed to eliminate duplicate files inside a info file

**Comment by Darakah** [comment]
@denemier ,

The `gallery-info` block is NOT only for the main gallery info page. It is a normal obsidian code block that can be placed in any md note to show information of an image as well as quickly access the info file of that image by right clicking the info block. 

Many people including myself use the `gallery-info` block outside the main gallery info files. As such, the `gallery-span-info` is not always present (even if present, having multiple gallery-info blocks in the same note becomes a problem). Although this can be annoying, you can easily edit the name from inside the main gallery by clicking the edit button. Removing the availability of the block for this simple problem is not very useful especially given how frequently one renames images.

**Comment by Darakah** [comment]
I can include an obsidian command, which would go through all info files inside the specified gallery info directory and update all the `gallery-info` imgPath to the same title as the image it is linked to. Would that solve your problem?

**Comment by denemier** [comment]
but how would this obsidian command be triggered, timeinterval is also bad practice, does obsidiab api let you know the file is changed for a linked mention?
but as you asked: changing **gallery-info imgPath** solves the issue, but doesn't it  also mean we can also add command after this to change the info files name as well like you offer to change imgPath
